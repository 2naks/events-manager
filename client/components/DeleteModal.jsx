import React, { Component } from 'react';
import axios from 'axios';
import { history } from '../routes';

/**
 * React component for delete modal
 */
export default class DeleteModal extends Component {
  /**
   * React's method to render react component.
   * Iterates array of events to render
   *
   * @return {object}
   */
  deleteObject = () => {
    if (this.props.item === 'event') {
      let eventId = this.props.objectId;
      axios({
        method: 'DELETE',
        url: `/api/v1/events/${eventId}`,
        headers: { 'x-access-token': localStorage.getItem('x-access-token') },
        withCredentials: true,
      })
        .then((response) => {
          history.push("/myevents");
        })
        .catch((err) => {

        });
    } else if (this.props.item === 'center') {
      let centerId = this.props.objectId;
      axios({
        method: 'DELETE',
        url: `/api/v1/centers/${centerId}`,
        headers: { 'x-access-token': localStorage.getItem('x-access-token') },
        withCredentials: true,
      })
        .then((response) => {
          this.props.show('addCenter'); // to trigger rerendering of AdminCenters
          return this.props.show('centers');
        })
        .catch((err) => {
          console.log(err);
        });
    }
  }

  /**
   * React's method to render react component.
   * Renders delete modal
   *
   * @return {object}
   */
  render() {
    return (
      <div className="modal fade"
        id={`${this.props.objectId}`} tabIndex="-1" role="dialog" aria-labelledby="11Label" aria-hidden="true">
        <div className="modal-dialog" role="document">
          <div className="modal-content">
            <div className="modal-header">
              <h6 className="modal-title" id="eventModalLabel">Delete {this.props.item}</h6>
              <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <div className="modal-body">
              <p className="text-center">Are you sure you want to delete this {this.props.item}?</p>
              <form />
            </div>
            <div className="modal-footer">
              <button type="button" className="btn btn-sm btn-secondary" data-dismiss="modal">Cancel</button>
              <button type="button" className="btn btn-sm btn-danger" data-dismiss="modal" onClick={this.deleteObject}>
                Delete {this.props.item}
              </button>
            </div>
          </div>
        </div>
      </div>

    );
  }
}
