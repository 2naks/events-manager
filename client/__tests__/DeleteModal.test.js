import DeleteModal from '../components/DeleteModal.jsx';

describe('DeleteModal Component', () => {
  global.localStorage = {
    getItem: (str) => 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NzUsInJvbGUiOiJhZG1pbiIsImlhdCI6MTUxMjEyMzIyMSwiZXhwIjoxNTEyMjA5NjIxfQ.IwykmkkET5VmBvTEyPPXfMOq611ITQaoh-9wV11mJgE'
  };

  const props = {
    objectId: "1"
  };

  it('it should render DeleteModal component', () => {
    const wrapper = shallow(<DeleteModal {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should handle onClick event', () => {
    const wrapper = mount(<DeleteModal {...props} />);
    let DeleteButton = wrapper.find('.btn-danger');
    DeleteButton.simulate('click');
  });
});
