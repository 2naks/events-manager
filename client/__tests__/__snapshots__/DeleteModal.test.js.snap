// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal Component it should render DeleteModal component 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="11Label"
  className="modal fade"
  id="1"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h6
          className="modal-title"
          id="eventModalLabel"
        >
          Delete 
        </h6>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p
          className="text-center"
        >
          Are you sure you want to delete this 
          ?
        </p>
        <form />
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-sm btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-sm btn-danger"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Delete 
        </button>
      </div>
    </div>
  </div>
</div>
`;
